drivers/include/lis2dh12.h

```diff
@@ -257,7 +338,100 @@ int lis2dh12_init(lis2dh12_t *dev, const lis2dh12_params_t *params);
 int lis2dh12_read(const lis2dh12_t *dev, int16_t *data);
 
 /**
- * @brief   Power on the given device
+ * @brief   Clear the LIS2DH12 memory, clears all sampled data
+ *
+ * @param[in] dev       device descriptor
+ *
+ * @return  LIS2DH12_OK on success
+ */
+int lis2dh12_clear_data(const lis2dh12_t *dev);
+
+/**
+ * @brief   Change device scale value
+ *
+ * @param[in] dev       device descriptor
+ * @param[in] scale     change to given scale value
+ *
+ * @return  LIS2DH12_OK on success
+ */
+int lis2dh12_set_scale(lis2dh12_t *dev, lis2dh12_scale_t scale);
+
+/**
+ * @brief   Change device sampling rate
+ *
+ * @param[in] dev       device descriptor
+ * @param[in] rate      change to given sampling rate
+ *
+ * @return  LIS2DH12_OK on success
+ */
+int lis2dh12_set_datarate(const lis2dh12_t *dev, lis2dh12_rate_t rate);
+
+/**
+ * @brief   Change device power mode
+ *
+ * @param[in] dev           device descriptor
+ * @param[in] powermode     change to given power mode
+ *
+ * @return  LIS2DH12_OK on success
+ */
+int lis2dh12_set_powermode(const lis2dh12_t *dev, lis2dh12_powermode_t powermode);
+
+/**
+ * @brief   Configures the high pass filter
+ *
+ * @param[in] dev       device descriptor
+ * @param[in] config    device high pass configuration
+ *
+ * @return  LIS2DH12_OK on success
+ */
+int lis2dh12_set_highpass(const lis2dh12_t *dev, const lis2dh12_highpass_t *config);
+
+/**
+ * @brief   Set the reference value to control the high-pass reference.
+ *          In LIS2DH12_HP_MODE_REFERENCE the reference value is used to filter data
+ *          on all axis. Subtracts reference value from acceleration.
+ *          Note: LSB changes according to LIS2DH12_SCALE
+ *
+ * @param[in] dev           device descriptor
+ * @param[in] reference     reference value [8 Bit]
```

Comment:
What do these values mean?
E.g. what is a reference of 23? 128?

Reply:
It also depence of the SCALE. So the LSB changes with the current SCALE.

Check 4.3.1.2 in https://www.st.com/resource/en/application_note/dm00365457-lis2dh12-mems-digital-output-motion-sensor-ultralowpower-highperformance-3axis-nano-accelerometer-stmicroelectronics.pdf